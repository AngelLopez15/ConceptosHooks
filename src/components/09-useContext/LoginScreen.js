import React, { useContext } from 'react'
import { UserContext } from './UserContext'

export const LoginScreen = () => {
    
    // Obteniendo la referencia al userContext
    const {user,setUser} = useContext(UserContext)

    return (
        <div>
            <h1>LoginScreen</h1>
            <hr />
            <pre>
                {JSON.stringify(user, null, 3)}
            </pre>
            <button className="btn btn-primary"
                onClick={()=>setUser({
                    id:123,
                    name:'Angel'
                })}
            >
                Login
            </button>
        </div>
    )
}
