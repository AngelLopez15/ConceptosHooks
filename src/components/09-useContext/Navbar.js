import React from 'react'
import { Link, NavLink } from 'react-router-dom'

export const Navbar = () => {
    // Navlink solo nos sirve para poner una clase y se muestre activado la ruta donde estamos
    return (
        <nav className="navbar navbar-expand-lg navbar-dark bg-dark">
            <Link className="navbar-brand" to="/">useContext</Link>
            <div className="navbar-nav">
                <NavLink exact activeClassName="active" className="nav-item nav-link" to="/">Home</NavLink>
                <NavLink exact activeClassName="active" className="nav-item nav-link" to="/login">Login</NavLink>
                <NavLink exact activeClassName="active" className="nav-item nav-link" to="/about">About</NavLink>
            </div>
        </nav>
    )
}
