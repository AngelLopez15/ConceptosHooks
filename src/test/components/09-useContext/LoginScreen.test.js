import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import { configure, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import { UserContext } from '../../../components/09-useContext/UserContext';
import { LoginScreen } from '../../../components/09-useContext/LoginScreen';

describe('Pruebas en el <LoginScreen />', () => {

  configure({adapter: new Adapter()});
  
  const setUser = jest.fn()

  const wrapper = mount(
    <UserContext.Provider
      value={{
        setUser
      }}
    >
      <LoginScreen />
    </UserContext.Provider>
  )

  test('Debe de mostrarse correctamente', () => {

    expect(toJson(wrapper)).toMatchSnapshot();
    
  })
  
  test('Debe de ejecutar el setUser con el argumento esperado', () => {
    wrapper.find('button').prop('onClick')()

    expect(setUser).toHaveBeenCalledWith({
      id:123,
      name:'Angel'
    })

  });


})
