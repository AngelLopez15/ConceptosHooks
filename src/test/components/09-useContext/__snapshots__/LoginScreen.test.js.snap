// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el <LoginScreen /> Debe de mostrarse correctamente 1`] = `
<LoginScreen>
  <div>
    <h1>
      LoginScreen
    </h1>
    <hr />
    <pre />
    <button
      className="btn btn-primary"
      onClick={[Function]}
    >
      Login
    </button>
  </div>
</LoginScreen>
`;
